### PR TITLE
Fixed issue with method called `new` being emitted as construct signature

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/55075

### CODE REVIEW

File: tests/baselines/reference/objectTypesIdentityWithConstructSignatures2.types [hunk 1]
```diff
@@ -32,8 +32,8 @@ var a: { new(x: Date): string }
 >x : Date
 
 var b = { new(x: RegExp) { return ''; } }; // not a construct signature, function called new
->b : { new(x: RegExp): string; }
->{ new(x: RegExp) { return ''; } } : { new(x: RegExp): string; }
+>b : { "new"(x: RegExp): string; }
```
Comment (jakebailey) [inline tests/baselines/reference/objectTypesIdentityWithConstructSignatures2.types:35]: This doesn't seem right... aren't all of these cases changed explicitly construct signatures that shouldn't be quoted?

Comment (Andarist) [reply tests/baselines/reference/objectTypesIdentityWithConstructSignatures2.types:35]: This one definitely isnt wrong. Notice how this is inferred~ from an object literal and that cant even have a construct signature - it’s just a method called `new`

Comment (jakebailey) [reply tests/baselines/reference/objectTypesIdentityWithConstructSignatures2.types:35]: Agh, yeah. I think I skimmed this PR on my phone on the weekend and had it in my head that there was something going on.

**Comment by jakebailey** [comment]
@typescript-bot test this
@typescript-bot test top100
@typescript-bot user test this
@typescript-bot run dt

**Comment by jakebailey** [comment]
@typescript-bot pack this

**Review by jakebailey — APPROVED** [review]
LGTM, though maybe @rbuckton may have some other trick that would be preferred over this particular fix.

**Comment by Andarist** [comment]
@jakebailey could you rephrase? from what I can tell both types baselines and tooltips are changed (and fixed) by this change

**Comment by jakebailey** [comment]
My question was "should this fix be in declaration emit", but I half convinced myself the answer was "no" because of tooltips and baselines.

Of course, the fix could be applied to both `declarations.ts` and `typeToTypeNode` or whatever that thing is called.

**Comment by jakebailey** [comment]
On second thought, this doesn't feel right; I think I found a better fix via `getPropertyNameNodeForSymbol` / `createPropertyNameNodeForIdentifierOrLiteral`.

Would you rather I send a new PR for this or push to your branch? (I am keeping the credit for the tests, of course; I just cloned this PR and then swapped the fix.)

**Comment by jakebailey** [comment]
I don't want to stomp on your branch, I'll send a different PR.

**Comment by jakebailey** [comment]
See: #55750
